src/Skybrud.ImagePicker/PropertyEditors/ImagePicker/ValueConverters/ImageValueConverter.cs

```diff
@@ -69,57 +67,52 @@ public class ImageValueConverter : PropertyValueConverterBase {
 
             // Initialize a collection for the items
             List<object> items = new List<object>();
-            
+
             // Determine the item value type
             Type valueType = propertyType.DataType.ConfigurationAs<ImagePickerConfiguration>()?.ValueType;
 
-            foreach (Udi udi in udis)  {
-                
+            foreach (Udi udi in udis) {
+
+                var canGetPublishedSnapshot = _publishedSnapshotAccessor.TryGetPublishedSnapshot(out var publishedSnapshotAccessor);
+
+                if (!canGetPublishedSnapshot)
+                    continue;
+
                 // Look up the media
-                IPublishedContent media = _publishedSnapshotAccessor.PublishedSnapshot.Media.GetById(udi);
+                IPublishedContent media = publishedSnapshotAccessor.Media.GetById(udi);
                 if (media == null) continue;
```

Comment:
Just two minor things:

- Code for getting the published snapshot could very well be outside the loop, as the result will be the same for each iteration
- `publishedSnapshotAccessor` should just be `publishedSnapshot` as it is the actual published snapshot, and not the accessor